### DISCUSSION TITLE
ExpressVPN Support?

### DISCUSSION THREAD

**Discussion by Delgrey** [qdm12/gluetun#517] [Ideas]
Is it possible to add ExpressVPN? They provide .ovpn files. There is a thread here with someone trying to get it working via custom settings, but it's unclear if they were successful. I've tried but am getting the container shut down with an error starting "ERROR cannot setup routing: cannot add IP rule..." Not sure if this is a config item I've missed.

**Comment by Delgrey** [comment]
Actually that error might have just been a permission settings in Synology Docker.

**Reply by qdm12** [reply]
Yes that's most likely the NET_ADMIN capability missing?

On the other hand, if you can provide a zip file of all openvpn files and attach it to #341 I can add support for it. Although new servers added by them won't be added automatically since their openvpn files are behind a login page AFAIK.

**Reply by Delgrey** [reply]
Thanks Quinten. Yeah the error was the NET_ADMIN setting. Unfortunately Synology Docker is very limited and there doesn't seem to be any way to include that. Setting it to elevated privileges seems to be the only solution for the GUI. On the plus side I know much more about docker-compose than I did 24 hours ago.

Happy to help. What would you need for the openvpn file? Can I redact the private key etc or do you actually need one that works? This key is generated against may account right? (Or is it just specific to the VPN node?) I wasn't sure if I also needed my generated user ID and password but I added them to the environment variables anyway.

They can probably supply a list of servers but since the openvpn file is specific to a server the ideal way would be a GUI interface for the docker container that allows you to upload your specific file like router firmware like OpenWRT etc do. Have you seen anything like that in a docker container?

Also - is there any way to securely script an nslookup for the server hostname? I'm assuming at some point they will rotate IP addresses and my config will stop working. (They do this to try and get around the Netflix blacklists I think)

Cheers.

**Reply by qdm12** [reply]
> What would you need for the openvpn file? Can I redact the private key etc or do you actually need one that works?

1. What's the name of that private key (I. e. `BEGIN RSA PRIVATE KEY`)? Is there a `auth-user-pass` in your config file?

> the ideal way would be a GUI interface for the docker container that allows you to upload your specific file

Yes I will eventually support that one day, but that's probably going to take a few months. I'm working on the http control server which will ultimately be used for a future UI.

> is there any way to securely script an nslookup for the server hostname

Not yet (to avoid leaking you're DNS resolving a certain vpn server), but subscribe to #404 which will address this.
